### ISSUE TITLE
feat: clone or duplicate the request

### ISSUE THREAD

**Issue by Nishchit14** [firecamp-dev/firecamp#53]
**Is your feature request related to a problem? Please describe.**
Currently, the user is not able to clone or duplicate the saved request.

**Describe the solution you'd like**
The request should be cloned quickly with a shortcut and option menu. The request should have the duplicate/clone option in the menu.

**Comment by pedrobruneli** [comment]
Hey @Nishchit14, can i work on this one?

**Comment by Nishchit14** [comment]
@pedrobruneli Yes you can, join discord's **contributors** channel to ask for any help.

**Comment by pedrobruneli** [comment]
I opened the issue #95, cant startup the repo

**Comment by Nishchit14** [comment]
@pedrobruneli it's solved, you can try with a new fork.

**Comment by pedrobruneli** [comment]
> @pedrobruneli it's solved, you can try with a new fork.

I synced the fork but stills the same error, looks like it's getting @firecamp/types from .pnpm and at that folder doesn't contains the 0.2.2 version

**Comment by Nishchit14** [comment]
@pedrobruneli I just temporarily unlinked the workspace types and used the npm types package for now and I checked it at my end. It's working fine. Can you please try now? please sync the main branch.

**Comment by bathini79** [comment]
@Nishchit14  Does it need to be here :
![image](https://github.com/firecamp-dev/firecamp/assets/116077457/62c32bfb-64dc-4243-a91a-097cb4f5857f)

**Comment by Nishchit14** [comment]
@bathini79 yes, you can. you can set the option in the menu if the request is rest-API only. I'll prepare the API for creating a duplicate request soon.

**Comment by bathini79** [comment]
Okay, duplicate inside the collection itself? Or Cross Collection as well?

**Comment by Nishchit14** [comment]
At same place, within same collection/folder.

**Comment by bathini79** [comment]
@Nishchit14   I Placed the Icon . Can you Help me with the API Strategy ? Do I Need To Just Hit The API and Upn Success Reload the API?
![image](https://github.com/firecamp-dev/firecamp/assets/116077457/62ff376b-7fa4-4778-8722-7bd357344d93)

**Comment by Nishchit14** [comment]
That's great, you can use mock promise as of now with success resolver. I
am yet to create the backend api. I will share it as soon as possible.

**Comment by bathini79** [comment]
Okay. Just one last question. In the response are you planning to give all the records again?

**Comment by Nishchit14** [comment]
I will return only one request information in the response. That's fine if
tree is not updated as of now. We can implement it once the API is ready.

**Comment by bathini79** [comment]
@Nishchit14  Okay , Here's my PR : https://github.com/firecamp-dev/firecamp/pull/105 . I have added as much as i have in control . I Don't Have Access to cloud-api .

**Comment by TrevorBenson** [comment]
Is there any status update available on this issue? I was about to open the same feature request and realized it already exists and has work underway. This is a very helpful feature when there are several sub-routes to add.

**Comment by Nishchit14** [comment]
@TrevorBenson I am currently shipping the desktop apps on all platforms, Once I'll complete this sprint I'll surely release the clone request feature. All desktop apps will be released by this Month.

**Comment by wheresfrank** [comment]
Has this been released? I don't see a duplicate option in the browser or desktop app.
